src/controller/ReadInteraction.h

```diff
@@ -63,7 +64,8 @@ CHIP_ERROR ReportAttribute(Messaging::ExchangeManager * exchangeMgr, EndpointId
         readParams.mpDataVersionFilterList    = dataVersionFilters.get();
         readParams.mDataVersionFilterListSize = 1;
     }
-    auto onDone = [](TypedReadAttributeCallback<DecodableAttributeType> * callback) { chip::Platform::Delete(callback); };
+    auto onDone = readParams.mOnDoneCb
+        ?: [](TypedReadAttributeCallback<DecodableAttributeType> * callback) { chip::Platform::Delete(callback); };
```

Comment:
Ideally, this code, which is what allocated `callback` would be responsible for deleting it, and the callback to the consumer would just take no arguments (but still be an std::function so consumers can pass in closures and whatnot).  So we would call the consumer callback, if there is one, then delete `callback` here.  Something like:

```
auto * onDoneCb = readParams.mOnDoneCb;
auto onDone = [onDoneCb](TypedReadAttributeCallback<DecodableAttributeType> * callback) 
{
    if (onDoneCb)
    {
        onDoneCb();
    }
    chip::Platform::Delete(callback);
};
```
or so.  And then consumers can just do whatever cleanup they actually need to do.

Reply:
Done